// @particle-network/authkit/dist/esm/jp-C4M2DYHQ.mjs
"use client";
import "./chunk-55PCA22M.mjs";

// src/locales/jp.json
var common = {
  confirm: "\u3092\u9078\u629E\u3057\u3066\u3001",
  still_confirm: "\u78BA\u8A8D",
  approve: "\u627F\u8A8D",
  cancel: "\u30AD\u30E3\u30F3\u30BB\u30EB",
  particle_auth: "Particle Auth",
  retry: "\u518D\u8A66\u884C",
  save: "\u4FDD\u5B58\uFF03\u30DB\u30BE\u30F3\uFF03",
  next: "\u3092\u9078\u629E\u3057\u3066\u3001",
  bind: "\u8A2D\u5B9A"
};
var login = {
  title: "\u30ED\u30B0\u30A4\u30F3\uFF03\u30ED\u30B0\u30A4\u30F3\uFF03",
  enter_email: "\u30E1\u30FC\u30EB\u30DC\u30C3\u30AF\u30B9",
  input_email_holder: "\u30E1\u30FC\u30EB\u30A2\u30C9\u30EC\u30B9\u3092\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044",
  login_with_phone: "\u643A\u5E2F\u756A\u53F7\u3067\u30ED\u30B0\u30A4\u30F3",
  login_with_email: "\u30E1\u30FC\u30EB\u30DC\u30C3\u30AF\u30B9\u3092\u4F7F\u7528\u3057\u305F\u30ED\u30B0\u30A4\u30F3",
  get_captcha: "\u8A8D\u8A3C\u30B3\u30FC\u30C9\u306E\u9001\u4FE1",
  enter_phone: "\u643A\u5E2F\u756A\u53F7",
  input_phone_holder: "\u643A\u5E2F\u756A\u53F7\u3092\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044",
  enter_captcha: "Please input your captcha",
  captcha_input_error: "\u8A8D\u8A3C\u30B3\u30FC\u30C9\u30A8\u30E9\u30FC\u3001\u518D\u8A66\u884C\u3057\u3066\u304F\u3060\u3055\u3044",
  send_again: "\u518D\u53D6\u5F97",
  select_country: "\u56FD\u3092\u9078\u629E",
  search_country_holder: "\u540D\u524D\u306B\u3088\u308B\u691C\u7D22",
  email_format_error: "\u30E1\u30FC\u30EB\u30DC\u30C3\u30AF\u30B9\u30D5\u30A9\u30FC\u30DE\u30C3\u30C8\u30A8\u30E9\u30FC",
  phone_format_error: "\u96FB\u8A71\u756A\u53F7\u306E\u5F62\u5F0F\u304C\u9593\u9055\u3063\u3066\u3044\u308B",
  enter_six_code: "6\u30D3\u30C3\u30C8\u8A8D\u8A3C\u30B3\u30FC\u30C9\u3092\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044",
  please_select_country: "\u56FD\u3092\u9078\u629E\u3057\u3066\u304F\u3060\u3055\u3044",
  success: "\u30ED\u30B0\u30A4\u30F3\u6210\u529F",
  welcome: "\u3088\u3046\u3053\u305D",
  not_login: "\u30ED\u30B0\u30A4\u30F3\u3057\u3066\u3044\u307E\u305B\u3093",
  return_to_the_app: "\u30A2\u30D7\u30EA\u30B1\u30FC\u30B7\u30E7\u30F3\u306B\u623B\u308B",
  invalid_code: "\u8A8D\u8A3C\u30B3\u30FC\u30C9\u304C\u7121\u52B9\u3067\u3059",
  email_address: "\u30E1\u30FC\u30EB\u30A2\u30C9\u30EC\u30B9",
  mobile_number: "\u643A\u5E2F\u756A\u53F7",
  please_send_again: "\u518D\u9001\u3057\u3066\u304F\u3060\u3055\u3044",
  email_verification_code: "\u30E1\u30FC\u30EB\u30DC\u30C3\u30AF\u30B9\u8A8D\u8A3C\u30B3\u30FC\u30C9",
  phone_verification_code: "\u643A\u5E2F\u96FB\u8A71\u8A8D\u8A3C\u30B3\u30FC\u30C9",
  to_continue: "\u30ED\u30B0\u30A4\u30F3\u5148{0}",
  or: "OR",
  social_login: "\u30ED\u30B0\u30A4\u30F3\u4E2D{0}",
  create_wallet: "\u30A6\u30A9\u30EC\u30C3\u30C8\u3092\u4F5C\u6210\u4E2D...",
  back: "\u623B\u308B",
  google: "Google",
  facebook: "Facebook",
  twitter: "Twitter",
  invalid_phone_number: "\u6709\u52B9\u306A\u96FB\u8A71\u756A\u53F7\u3092\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044\uFF01",
  passkeys: "PassKeys",
  passkeys_auth_error: "\u3053\u306EPassKeys\u306B\u95A2\u9023\u3059\u308B\u30A2\u30AB\u30A6\u30F3\u30C8\u306F\u3042\u308A\u307E\u305B\u3093\u3002",
  logging_you_in: "\u30ED\u30B0\u30A4\u30F3\u4E2D",
  request_failed: "\u30EA\u30AF\u30A8\u30B9\u30C8\u304C\u5931\u6557\u3057\u307E\u3057\u305F",
  something_wrong: "\u30A8\u30E9\u30FC\u304C\u767A\u751F\u3057\u307E\u3057\u305F\u3002\u5F8C\u3067\u3082\u3046\u4E00\u5EA6\u4E0A\u306E\u30DC\u30BF\u30F3\u3092\u30AF\u30EA\u30C3\u30AF\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  google_connected: "Google\u306B\u63A5\u7D9A\u3057\u307E\u3057\u305F",
  good_to_go: "\u3059\u3079\u3066\u6E96\u5099\u304C\u6574\u3044\u307E\u3057\u305F\u3001\u7D9A\u884C\u3067\u304D\u307E\u3059\uFF01",
  request_cancelled: "\u30EA\u30AF\u30A8\u30B9\u30C8\u304C\u30AD\u30E3\u30F3\u30BB\u30EB\u3055\u308C\u307E\u3057\u305F",
  you_cancelled: "\u30EA\u30AF\u30A8\u30B9\u30C8\u304C\u30AD\u30E3\u30F3\u30BB\u30EB\u3055\u308C\u307E\u3057\u305F\u3002\u4E0A\u306E\u30DC\u30BF\u30F3\u3092\u30AF\u30EA\u30C3\u30AF\u3057\u3066\u518D\u8A66\u884C\u3057\u3066\u304F\u3060\u3055\u3044\u3002"
};
var sign = {
  wallet: "\u8CA1\u5E03",
  signature_message: "\u30E1\u30C3\u30BB\u30FC\u30B8\u7F72\u540D",
  signature_title: "\u3053\u306E\u30A2\u30D7\u30EA\u30B1\u30FC\u30B7\u30E7\u30F3\u306F\u3042\u306A\u305F\u306E\u7F72\u540D\u3092\u53D6\u5F97\u3059\u308B\u3088\u3046\u306B\u8981\u6C42\u3057\u307E\u3059\u3002",
  sign_transaction: "\u53D6\u5F15\u7F72\u540D",
  sign_but: "\u7F72\u540D\u306E\u307F\u3092Solana\u30CD\u30C3\u30C8\u30EF\u30FC\u30AF\u306B\u9001\u4FE1\u3057\u307E\u305B\u3093\u3002",
  send_transaction: "\u53D6\u5F15\u3092\u9001\u308B",
  approve_and: "{0}\u30CD\u30C3\u30C8\u30EF\u30FC\u30AF\u306B\u53D6\u5F15\u3092\u627F\u8A8D\u3057\u3066\u9001\u4FE1\u3057\u307E\u3059\u3002",
  sign_typed_data: "\u7F72\u540D\u30BF\u30A4\u30D7\u30C7\u30FC\u30BF",
  network_fee: "\u30CD\u30C3\u30C8\u30EF\u30FC\u30AF\u8CBB\u7528",
  evm_fee: "Gas\uFF08\u63A8\u5B9A\uFF09",
  show_more: "\u30D7\u30EC\u30BC\u30F3\u30C6\u30FC\u30B7\u30E7\u30F3",
  show_less: "\u3084\u3081\u308B",
  full_name: "\u30D5\u30EB\u30CD\u30FC\u30E0",
  user_id: "\u30E6\u30FC\u30B6ID",
  estimated_balance_change: "\u4E88\u6E2C\u6B8B\u9AD8\u306E\u5909\u66F4",
  please_add: "\u305B\u3081\u3066\u5897\u3084\u3057\u3066\u304F\u3060\u3055\u3044",
  inner_instruction: "\u5185\u90E8\u6458\u8981",
  program_id: "\u30D7\u30ED\u30B0\u30E9\u30E0ID",
  data: "\u30C7\u30FC\u30BF\uFF03\u30C7\u30FC\u30BF\uFF03",
  details: "\u8A73\u7D30",
  from: "\u9001\u4FE1\u5148",
  to: "\u53D7\u4FE1\u30A2\u30C9\u30EC\u30B9",
  max_fee: "\u6700\u5927\u8CBB\u7528\uFF1A",
  total: "\u306E\u5408\u8A08\u3068\u306A\u308B",
  max_amount: "\u6700\u5927\u6570",
  transaction_action: "\u53D6\u5F15\u884C\u70BA",
  swap: "\u30B9\u30EF\u30C3\u30D7",
  for: "\u3042\u3052\u308B",
  on: "\u305D\u306E",
  hex_data: "Hex\u30C7\u30FC\u30BF",
  function_type: "\u30E1\u30BD\u30C3\u30C9\uFF1A",
  nonce: "Nonce",
  edit_priority: "\u901F\u5EA6\u3092\u8ABF\u6574\u3059\u308B",
  advanced_options: "\u8A73\u7D30\u30AA\u30D7\u30B7\u30E7\u30F3",
  gas_limit: "Gas\u5236\u9650",
  send_max: "\u6700\u5927\u9001\u4FE1",
  max_priority_fee: "\u3055\u3044\u3060\u3044\u304B\u305D\u304F\u3072",
  estimate: "\u898B\u7A4D\u3082\u308A\uFF1A",
  max_fee_fee: "\u3055\u3044\u3060\u3044\u3072\u3088\u3046",
  gas_limit_must: "Gas\u306F\u5C11\u306A\u304F\u3068\u3082{0}\u306B\u5236\u9650\u3055\u308C\u3066\u3044\u308B\u3002",
  pr_fee_is_low: "\u6700\u5927\u52A0\u901F\u6599\u91D1\u306F\u73FE\u5728\u306E\u30CD\u30C3\u30C8\u30EF\u30FC\u30AF\u6761\u4EF6\u3092\u4E0B\u56DE\u3063\u3066\u3044\u308B\u3002",
  pr_fee_is_high: "\u512A\u5148\u5EA6\u306E\u6700\u5927\u8CBB\u7528\u306F\u5FC5\u8981\u306A\u8CBB\u7528\u3088\u308A\u3082\u9AD8\u3044\u3002",
  fee_is_low: "\u6700\u5927\u6599\u91D1\u306F\u73FE\u5728\u306E\u30CD\u30C3\u30C8\u30EF\u30FC\u30AF\u6761\u4EF6\u3092\u4E0B\u56DE\u3063\u3066\u3044\u307E\u3059\u3002",
  fee_is_high: "\u6700\u9AD8\u8CBB\u7528\u306F\u5FC5\u8981\u8CBB\u7528\u3088\u308A\u9AD8\u3044\u3002",
  fee_low_priority: "\u6700\u9AD8\u6599\u91D1\u306F\u6700\u9AD8\u512A\u5148\u6599\u91D1\u3092\u4E0B\u56DE\u308B\u3053\u3068\u306F\u3067\u304D\u307E\u305B\u3093\u3002",
  gas_low: "\u9045\u3044",
  gas_medium: "\u4E2D",
  gas_high: "\u306F\u3084\u3044",
  gas_custom: "\u30AB\u30B9\u30BF\u30E0\uFF03",
  refresh_in: "\u66F4\u65B0",
  gas_fee_cannot: "\u30AC\u30B9\u6599\u91D1\u306F\u3057\u3070\u3089\u304F\u8A66\u7B97\u3067\u304D\u306A\u3044\u3002\u5951\u7D04\u306B\u30A8\u30E9\u30FC\u304C\u3042\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u3001\u53D6\u5F15\u304C\u5931\u6557\u3059\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059\u3002",
  risk_found: "\u30EA\u30B9\u30AF\u306E\u691C\u51FA",
  risk_hint_title: "\u7F72\u540D\u691C\u51FA",
  risk_identified: "\u4EE5\u4E0B\u306E\u30EA\u30B9\u30AF\u304C\u767A\u898B\u3055\u308C\u307E\u3057\u305F\u3002",
  transaction_may_fail: "\u6CE8\u610F\u4E8B\u9805",
  security_risk: "\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u30EA\u30B9\u30AF"
};
var error = {
  webview_unsupported: "Webview\u3067\u958B\u304F\u3053\u3068\u306F\u30B5\u30DD\u30FC\u30C8\u3055\u308C\u3066\u3044\u307E\u305B\u3093",
  params_error: "\u30D1\u30E9\u30E1\u30FC\u30BF\u30A8\u30E9\u30FC",
  not_login_error: "\u30E6\u30FC\u30B6\u30FC\u304C\u30ED\u30B0\u30A4\u30F3\u3057\u3066\u3044\u307E\u305B\u3093",
  unknown: "\u4E0D\u660E\u306A\u30A8\u30E9\u30FC",
  server_20102: "\u30B3\u30FC\u30C9\u4F1D\u9001\u901F\u5EA6\u5236\u9650",
  server_50003: "\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u96FB\u5B50\u30E1\u30FC\u30EB\u306F\u30ED\u30B0\u30A4\u30F3\u96FB\u5B50\u30E1\u30FC\u30EB\u3068\u540C\u3058\u306B\u3067\u304D\u307E\u305B\u3093",
  server_50004: "\u5B89\u5168\u306A\u96FB\u8A71\u756A\u53F7\u306F\u767B\u9332\u96FB\u8A71\u756A\u53F7\u3068\u540C\u3058\u306B\u3059\u308B\u3053\u3068\u306F\u3067\u304D\u307E\u305B\u3093",
  server_50101: "\u5143\u306E\u30D1\u30B9\u30EF\u30FC\u30C9\u306E\u691C\u8A3C\u306B\u5931\u6557\u3057\u307E\u3057\u305F",
  server_50104: "\u65B0\u3057\u3044\u30D1\u30B9\u30EF\u30FC\u30C9\u306F\u4EE5\u524D\u306E\u30D1\u30B9\u30EF\u30FC\u30C9\u3068\u540C\u3058\u306B\u3059\u308B\u3053\u3068\u306F\u3067\u304D\u307E\u305B\u3093",
  server_10005: "\u30ED\u30B0\u30A4\u30F3\u304C\u7121\u52B9\u306B\u306A\u308A\u307E\u3057\u305F\u3002\u518D\u30ED\u30B0\u30A4\u30F3\u3057\u3066\u304F\u3060\u3055\u3044",
  server_email_20109: "\u3053\u306E\u30E1\u30FC\u30EB\u30DC\u30C3\u30AF\u30B9\u306FParticle Network\u306B\u767B\u9332\u3055\u308C\u3066\u3044\u307E\u3059\u3002\u5225\u306E\u30E1\u30FC\u30EB\u30DC\u30C3\u30AF\u30B9\u3092\u4F7F\u7528\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  server_phone_20109: "\u3053\u306E\u643A\u5E2F\u96FB\u8A71\u756A\u53F7\u306FParticle Network\u306B\u767B\u9332\u3055\u308C\u3066\u3044\u307E\u3059\u3002\u5225\u306E\u643A\u5E2F\u756A\u53F7\u3092\u4F7F\u7528\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  server_google_20109: "\u3053\u306EGoogle\u30A2\u30AB\u30A6\u30F3\u30C8\u306FParticle Network\u306B\u767B\u9332\u3055\u308C\u3066\u3044\u307E\u3059\u3002\u5225\u306E\u30A2\u30AB\u30A6\u30F3\u30C8\u3092\u4F7F\u7528\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  server_facebook_20109: "\u3053\u306EFacebook\u30A2\u30AB\u30A6\u30F3\u30C8\u306FParticle Network\u306B\u767B\u9332\u3055\u308C\u3066\u3044\u307E\u3059\u3002\u5225\u306E\u30A2\u30AB\u30A6\u30F3\u30C8\u3092\u4F7F\u7528\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  server_twitter_20109: "\u3053\u306ETwitter\u30A2\u30AB\u30A6\u30F3\u30C8\u306FParticle Network\u306B\u767B\u9332\u3055\u308C\u3066\u3044\u307E\u3059\u3002\u5225\u306E\u30A2\u30AB\u30A6\u30F3\u30C8\u3092\u4F7F\u7528\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  server_20112: "CAPTCHA \u691C\u8A3C\u306B\u5408\u683C\u3057\u307E\u305B\u3093\u3067\u3057\u305F\u3002 \u5F8C\u3067\u3082\u3046\u4E00\u5EA6\u3084\u308A\u76F4\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  wallet_not_activated: "\u30A6\u30A9\u30EC\u30C3\u30C8\u306E\u30A8\u30F3\u30C8\u30EA\u30FC\u306F\u30A2\u30AF\u30C6\u30A3\u30D6\u3067\u306F\u3042\u308A\u307E\u305B\u3093\uFF1B\u958B\u767A\u8005\u306B\u9023\u7D61\u3057\u3066\u30A2\u30AF\u30C6\u30A3\u30D6\u5316\u3057\u3066\u304F\u3060\u3055\u3044\u3002"
};
var program = {
  unknown: "Unknown",
  "transfer-sol": "Transfer SOL",
  "transfer-nft": "Transfer NFT",
  "transfer-token": "Transfer Token",
  "create-token-account": "Create Token Account",
  "system-create": "System Create",
  "system-assign": "System Assign",
  "system-transfer": "System Transfer",
  "system-create-with-seed": "System Create With Seed",
  "system-advance-nonce-account": "System Advance Nonce Account",
  "system-withdraw-nonce-account": "System Withdraw Nonce Account",
  "system-initialize-nonce-account": "System Initialize Nonce Account",
  "system-authorize-nonce-account": "System Authorize Nonce Account",
  "system-allocate": "System Allocate",
  "system-allocate-with-seed": "System Allocate With Seed",
  "system-assign-with-seed": "System Assign With Seed",
  "system-transfer-with-seed": "System Transfer With Seed",
  "system-upgrade-nonce-account": "System Upgrade Nonce Account",
  "vote-initialize-account": "Vote Initialize Account",
  "vote-authorize": "Vote Authorize",
  "vote-withdraw": "Vote Withdraw",
  "stake-initialize": "Stake Initialize",
  "stake-authorize": "Stake Authorize",
  "stake-delegate": "Stake Delegate",
  "stake-split": "Stake Split",
  "stake-withdraw": "Stake Withdraw",
  "stake-deactivate": "Stake Deactivate",
  "stake-merge": "Stake Merge",
  "stake-authorize-with-seed": "Stake Authorize With Seed",
  "token-initialize-mint": "Token Initialize Mint",
  "token-initialize-account": "Token Initialize Account",
  "token-initialize-multisig": "Token Initialize Multisig",
  "token-transfer": "Token Transfer",
  "token-approve": "Token Approve",
  "token-revoke": "Token Revoke",
  "token-set-authority": "Token Set Authority",
  "token-mint-to": "Token Mint To",
  "token-burn": "Token Burn",
  "token-close-account": "Token Close Account",
  "token-freeze-account": "Token Freeze Account",
  "token-thaw-account": "Token Thaw Account",
  "token-transfer-checked": "Token Transfer Checked",
  "token-approve-checked": "Token Approve Checked",
  "token-mint-to-checked": "Token Mint To Checked",
  "token-burn-checked": "Token Burn Checked",
  "token-initialize-account-2": "Token Initialize Account 2",
  "token-sync-native": "Token Sync Native",
  "token-initialize-account-3": "Token Initialize Account 3",
  "token-initialize-multisig-2": "Token Initialize Multisig 2",
  "token-initialize-mint-2": "Token Initialize Mint 2",
  "associated-token-create": "Associated Token Create",
  "token-swap-initialize-swap": "Token Swap Initialize Swap",
  "token-swap-swap": "Token Swap Swap",
  "token-swap-deposit": "Token Swap Deposit",
  "token-swap-withdraw": "Token Swap Withdraw",
  "token-swap-deposit-single-token-type-exact-amount-in": "Token Swap Deposit Single Token Type Exact Amount In",
  "token-swap-withdraw-single-token-type-exact-amount-out": "Token Swap Withdraw Single Token Type Exact Amount Out",
  "memo-create": "Memo Create",
  "metadata-create-metadata-account": "Create Metadata Account",
  "metadata-update-metadata-account": "Update Metadata Account",
  "metadata-deprecated-create-master-edition": "Deprecated Create Master Edition",
  "metadata-deprecated-mint-new-edition-from-master-edition-via-printing-token": "Deprecated Mint New Edition From Master Edition Via Printing Token",
  "metadata-update-primary-sale-happened-via-token": "Update Primary Sale Happened Via Token",
  "metadata-deprecated-set-reservation-list": "Deprecated Set Reservation List",
  "metadata-deprecated-create-reservation-list": "Deprecated Create Reservation List",
  "metadata-sign-metadata": "Sign Metadata",
  "metadata-deprecated-mint-printing-tokens-via-token": "Deprecated Mint Printing Tokens Via Token",
  "metadata-deprecated-mint-printing-tokens": "Deprecated Mint Printing Tokens",
  "metadata-create-master-edition": "Create Master Edition",
  "metadata-mint-new-edition-from-master-edition-via-token": "Mint New Edition From Master Edition Via Token",
  "metadata-convert-master-edition-v1-to-v2": "Convert Master Edition V1 To V2",
  "metadata-mint-new-edition-from-master-edition-via-vault-proxy": "Mint New Edition From Master Edition Via Vault Proxy",
  "metadata-puff-metadata": "Puff Metadata",
  "metadata-update-metadata-account-v2": "Update Metadata Account V2",
  "metadata-create-metadata-account-v2": "Create Metadata Account V2",
  "metadata-create-master-edition-v3": "Create Master Edition V3",
  "metadata-verify-collection": "Verify Collection",
  "metadata-utilize": "Utilize",
  "metadata-approve-use-authority": "Approve Use Authority",
  "metadata-revoke-use-authority": "Revoke Use Authority",
  "metadata-unverify-collection": "Unverify Collection",
  "metadata-approve-collection-authority": "Approve Collection Authority",
  "metadata-revoke-collection-authority": "Revoke Collection Authority",
  "metadata-set-and-verify-collection": "Set And Verify Collection",
  "metadata-freeze-delegated-account": "Freeze Delegated Account",
  "metadata-thaw-delegated-account": "Thaw Delegated Account",
  "metadata-remove-creator-verification": "Remove Creator Verification",
  "metadata-burn-nft": "Burn Nft",
  "metadata-verify-sized-collection-item": "Verify Sized Collection Item",
  "metadata-unverify-sized-collection-item": "Unverify Sized Collection Item",
  "metadata-set-and-verify-sized-collection-item": "Set And Verify Sized Collection Item",
  "metadata-create-metadata-account-v3": "Create Metadata Account V3",
  "metadata-set-collection-size": "Set Collection Size",
  "metadata-set-token-standard": "Set Token Standard",
  "vault-init-vault": "Init Vault",
  "vault-add-token-to-inactive-vault": "Add Token To Inactive Vault",
  "vault-activate-vault": "Activate Vault",
  "vault-combine-vault": "Combine Vault",
  "vault-redeem-shares": "Redeem Shares",
  "vault-withdraw-token-from-safety-deposit-box": "Withdraw Token From Safety Deposit Box",
  "vault-mint-fractional-shares": "Mint Fractional Shares",
  "vault-withdraw-shares-from-treasury": "Withdraw Shares From Treasury",
  "vault-add-shares-to-treasury": "Add Shares To Treasury",
  "vault-update-external-price-account": "Update External Price Account",
  "vault-set-authority": "Set Authority",
  "metaplex-redeem-bid": "Metaplex Redeem Bid",
  "metaplex-redeem-full-rights-transfer-bid": "Metaplex Redeem Full Rights Transfer Bid",
  "metaplex-start-auction": "Metaplex Start Auction",
  "metaplex-claim-bid": "Metaplex Claim Bid",
  "metaplex-empty-payment-account": "Metaplex Empty Payment Account",
  "metaplex-set-store": "Metaplex Set Store",
  "metaplex-set-whitelisted-creator": "Metaplex Set Whitelisted Creator",
  "metaplex-redeem-unused-winning-config-items-as-auctioneer": "Metaplex Redeem Unused Winning Config Items As Auctioneer",
  "metaplex-redeem-printing-v2-bid": "Metaplex Redeem Printing V2 Bid",
  "metaplex-init-auction-manager-v2": "Metaplex Init Auction Manager V2",
  "metaplex-validate-safety-deposit-box-v2": "Metaplex Validate Safety Deposit Box V2",
  "metaplex-redeem-participation-bid-v3": "Metaplex Redeem Participation Bid V3",
  "metaplex-end-auction": "Metaplex End Auction",
  "metaplex-set-store-v2": "Metaplex Set Store V2",
  "auction-cancel-bid": "Auction Cancel Bid",
  "auction-create-auction": "Auction Create Auction",
  "auction-set-auction-authority": "Auction Set Auction Authority",
  "auction-place-bid": "Auction Place Bid",
  "auction-create-auction-v2": "Auction Create Auction V2",
  "auction-house-initialize-candy-machine": "Auction House Initialize Candy Machine",
  "auction-house-update-candy-machine": "Auction House Update Candy Machine",
  "auction-house-update-authority": "Auction House Update Authority",
  "auction-house-add-config-lines": "Auction House Add Config Lines",
  "auction-house-set-collection": "Auction House Set Collection",
  "auction-house-remove-collection": "Auction House Remove Collection",
  "auction-house-mint-nft": "Auction House Mint Nft",
  "auction-house-set-collection-during-mint": "Auction House Set Collection During Mint",
  "auction-house-withdraw-funds": "Auction House Withdraw Funds",
  "auction-house-withdraw-from-fee": "Auction House Withdraw From Fee",
  "auction-house-withdraw-from-treasury": "Auction House Withdraw From Treasury",
  "auction-house-update-auction-house": "Auction House Update Auction House",
  "auction-house-create-auction-house": "Auction House Create Auction House",
  "auction-house-buy": "Auction House Buy",
  "auction-house-auctioneer-buy": "Auction House Auctioneer Buy",
  "auction-house-public-buy": "Auction House Public Buy",
  "auction-house-auctioneer-public-buy": "Auction House Auctioneer Public Buy",
  "auction-house-cancel": "Auction House Cancel",
  "auction-house-auctioneer-cancel": "Auction House Auctioneer Cancel",
  "auction-house-deposit": "Auction House Deposit",
  "auction-house-auctioneer-deposit": "Auction House Auctioneer Deposit",
  "auction-house-execute-sale": "Auction House Execute Sale",
  "auction-house-execute-partial-sale": "Auction House Execute Partial Sale",
  "auction-house-auctioneer-execute-sale": "Auction House Auctioneer Execute Sale",
  "auction-house-auctioneer-execute-partial-sale": "Auction House Auctioneer Execute Partial Sale",
  "auction-house-sell": "Auction House Sell",
  "auction-house-auctioneer-sell": "Auction House Auctioneer Sell",
  "auction-house-withdraw": "Auction House Withdraw",
  "auction-house-auctioneer-withdraw": "Auction House Auctioneer Withdraw",
  "auction-house-close-escrow-account": "Auction House Close Escrow Account",
  "auction-house-delegate-auctioneer": "Auction House Delegate Auctioneer",
  "auction-house-update-auctioneer": "Auction House Update Auctioneer",
  "auction-house-print-listing-receipt": "Auction House Print Listing Receipt",
  "auction-house-cancel-listing-receipt": "Auction House Cancel Listing Receipt",
  "auction-house-print-bid-receipt": "Auction House Print Bid Receipt",
  "auction-house-cancel-bid-receipt": "Auction House Cancel Bid Receipt",
  "auction-house-print-purchase-receipt": "Auction House Print Purchase Receipt",
  "jupiter-mercurial-exchange": "Mercurial Exchange",
  "jupiter-saber-exchange": "Saber Exchange",
  "jupiter-saber-swap": "Saber Swap",
  "jupiter-saber-add-decimals-deposit": "Saber Add Decimals Deposit",
  "jupiter-saber-add-decimals-withdraw": "Saber Add Decimals Withdraw",
  "jupiter-sencha-exchange": "Sencha Exchange",
  "jupiter-serum-swap": "Serum Swap",
  "jupiter-token-swap": "Token Swap",
  "jupiter-step-token-swap": "Step Token Swap",
  "jupiter-cropper-token-swap": "Cropper Token Swap",
  "jupiter-raydium-swap": "Raydium Swap",
  "jupiter-raydium-swap-v2": "Raydium Swap V2",
  "jupiter-aldrin-swap": "Aldrin Swap",
  "jupiter-aldrin-v2-swap": "Aldrin V2 Swap",
  "jupiter-crema-token-swap": "Crema Token Swap",
  "jupiter-lifinity-token-swap": "Lifinity Token Swap",
  "jupiter-cykura-swap": "Cykura Swap",
  "jupiter-whirlpool-swap": "Whirlpool Swap",
  "jupiter-whirlpool-swap-exact-output": "Whirlpool Swap Exact Output",
  "jupiter-marinade-finance-deposit": "Marinade Finance Deposit",
  "jupiter-marinade-finance-liquid-unstake": "Marinade Finance Liquid Unstake",
  "jupiter-risk-check-and-fee": "Risk Check And Fee",
  "jupiter-initialize-token-ledger": "Initialize Token Ledger",
  "jupiter-set-token-ledger": "Set Token Ledger",
  "jupiter-create-open-orders": "Create Open Orders"
};
var new2 = {
  confirmation: "\u9001\u4FE1\u78BA\u8A8D",
  from: "\u9001\u4FE1",
  you: "\u3042\u306A\u305F",
  to: "\u53D7\u5165",
  network_fee: "\u30CD\u30C3\u30C8\u30EF\u30FC\u30AF\u8CBB\u7528",
  less: "\u3084\u3081\u308B",
  details: "\u8A73\u7D30",
  token_per: "\u652F\u6255\u627F\u8A8D",
  edit_approve: "\u627F\u8A8D\u91D1\u984D\u306E\u7DE8\u96C6",
  by_granting: "*\u627F\u8A8D\u306B\u540C\u610F\u3059\u308B\u3068\u3001\u3053\u306E\u30A2\u30C9\u30EC\u30B9\u306F\u627F\u8A8D\u3055\u308C\u305F\u7BC4\u56F2\u5185\u3067\u8CC7\u91D1\u3092\u4F7F\u7528\u3067\u304D\u307E\u3059\u3002",
  amount: "\u6570\u91CF",
  please_enter: "\u6709\u52B9\u306A\u91D1\u984D\u3092\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044",
  the_amount: "\u91D1\u984D\u306F0\u3088\u308A\u5927\u304D\u304F\u306A\u3051\u308C\u3070\u306A\u308A\u307E\u305B\u3093",
  sending: "\u9001\u4FE1",
  buy_nft: "NFT\u306E\u8CFC\u5165",
  cancel_nft_listing: "NFT\u306E\u8CA9\u58F2\u4E2D\u6B62",
  transaction_pending: "\u3042\u306A\u305F\u304C\u30DA\u30F3\u3092\u6301\u3063\u3066\u3044\u308B\u524D\u306E\u53D6\u5F15\u306F\u307E\u3060\u78BA\u8A8D\u3055\u308C\u3066\u3044\u307E\u305B\u3093\u3002\u901F\u3084\u304B\u306B\u51E6\u7406\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  continue: "\u7F72\u540D\u306E\u7D99\u7D9A",
  process_now: "\u73FE\u5728\u51E6\u7406",
  link_account_hint: "Particle\u30A6\u30A9\u30EC\u30C3\u30C8\u306B\u30ED\u30B0\u30A4\u30F3\u3059\u308B\u306B\u306F\u3001{0}\u3092\u4F7F\u7528\u3057\u307E\u3059\u3002",
  bind_login_account_success: "\u30ED\u30B0\u30A4\u30F3\u30A2\u30AB\u30A6\u30F3\u30C8\u306E\u30D0\u30A4\u30F3\u30C9\u306B\u6210\u529F\u3057\u307E\u3057\u305F",
  bind_phooe_is_security_phone: "\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u30A2\u30AB\u30A6\u30F3\u30C8\u306F\u30ED\u30B0\u30A4\u30F3\u30A2\u30AB\u30A6\u30F3\u30C8\u3068\u3057\u3066\u4F7F\u7528\u3067\u304D\u307E\u305B\u3093",
  bind_email_is_security_email: "\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u30A2\u30AB\u30A6\u30F3\u30C8\u306F\u30ED\u30B0\u30A4\u30F3\u30A2\u30AB\u30A6\u30F3\u30C8\u3068\u3057\u3066\u4F7F\u7528\u3067\u304D\u307E\u305B\u3093",
  transaction_activity: "\u30C8\u30E9\u30F3\u30B6\u30AF\u30B7\u30E7\u30F3",
  account_security: "\u30A2\u30AB\u30A6\u30F3\u30C8\u3068\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3",
  confirm_connection: "\u63A5\u7D9A\u306E\u78BA\u8A8D",
  transaction_pending_v2: "\u672A\u51E6\u7406\u306E\u53D6\u5F15\u304C\u3042\u308A\u307E\u3059\u306E\u3067\u3001\u901F\u3084\u304B\u306B\u51E6\u7406\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  duplicate_email_bindings: "\u30ED\u30B0\u30A4\u30F3\u30E1\u30FC\u30EB\u30DC\u30C3\u30AF\u30B9\u3092\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u30E1\u30FC\u30EB\u30DC\u30C3\u30AF\u30B9\u3068\u540C\u3058\u306B\u3059\u308B\u3053\u3068\u306F\u3067\u304D\u307E\u305B\u3093\u3002",
  duplicate_phonel_bindings: "\u30ED\u30B0\u30A4\u30F3\u3057\u305F\u643A\u5E2F\u756A\u53F7\u306F\u3001\u30BB\u30AD\u30E5\u30A2\u306A\u643A\u5E2F\u756A\u53F7\u3068\u540C\u3058\u306B\u3059\u308B\u3053\u3068\u306F\u3067\u304D\u307E\u305B\u3093",
  click_to_link_with_google: "\u30EA\u30F3\u30AF\u3092\u30AF\u30EA\u30C3\u30AF\u3057\u3066Google",
  click_to_link_with_facebook: "\u30EA\u30F3\u30AF\u3092\u30AF\u30EA\u30C3\u30AF\u3057\u3066Facebook",
  click_to_link_with_twitter: "\u30EA\u30F3\u30AF\u3092\u30AF\u30EA\u30C3\u30AF\u3057\u3066Twitter",
  opening_facebook: "Facebook\u3092\u958B\u3044\u3066\u3044\u307E\u3059",
  opening_google: "Google\u3092\u958B\u3044\u3066\u3044\u307E\u3059",
  opening_twitter: "Twitter\u3092\u958B\u3044\u3066\u3044\u307E\u3059",
  nft_listing: "\u30EA\u30B9\u30C8NFT",
  nft_price: "NFT \u4FA1\u683C",
  price: "\u4FA1\u683C",
  seller_earnings: "\u58F2\u308A\u624B\u306E\u53CE\u5165",
  others: "\u4ED6\u306E",
  start_time: "\u958B\u59CB\u6642\u9593",
  end_time: "\u7D42\u4E86\u6642\u9593",
  seller: "\u58F2\u308A\u624B",
  infinite: "\u671F\u9650\u5207\u308C\u306B\u306A\u308B\u3053\u3068\u306F\u3042\u308A\u307E\u305B\u3093",
  manage_devices: "\u7BA1\u7406\u88C5\u7F6E",
  manage_devices_tips: "\u6700\u8FD1\u30ED\u30B0\u30A4\u30F3\u3057\u305F\u30C7\u30D0\u30A4\u30B9: \u3053\u306E\u30EA\u30B9\u30C8\u304B\u3089\u4EFB\u610F\u306E\u30A2\u30AF\u30BB\u30B9\u8A8D\u8A3C\u3092\u524A\u9664\u3067\u304D\u307E\u3059\u304C\u3001\u524A\u9664\u3055\u308C\u305F\u30A2\u30D7\u30EA\u306B\u518D\u5EA6\u30ED\u30B0\u30A4\u30F3\u3059\u308B\u306B\u306F\u3001\u518D\u5EA6\u672C\u4EBA\u78BA\u8A8D\u304C\u5FC5\u8981\u3067\u3059\u3002",
  current: "\u4F7F\u3063\u3066\u3044\u307E\u3059",
  app: "\u30A2\u30D7\u30EA",
  device: "\u88C5\u7F6E",
  authorized_time: "\u627F\u8A8D\u6642\u9593",
  delete_access: "\u627F\u8A8D\u3092\u524A\u9664\u3059\u308B",
  delete_access_to_n: "{0}({1}) \u306E\u627F\u8A8D\u306F\u524A\u9664\u3055\u308C\u307E\u3059",
  authorization: "\u8A8D\u53EF\u3055\u308C\u305F",
  delete: "\u6D88\u53BB",
  authorization_tips: "\u3053\u306E\u30EA\u30B9\u30C8\u304B\u3089\u4EFB\u610F\u306E\u30A2\u30AF\u30BB\u30B9\u8A8D\u8A3C\u3092\u524A\u9664\u3067\u304D\u307E\u3059\u304C\u3001\u524A\u9664\u3055\u308C\u305F\u30A2\u30D7\u30EA\u306B\u518D\u5EA6\u30ED\u30B0\u30A4\u30F3\u3059\u308B\u306B\u306F\u3001\u518D\u5EA6\u672C\u4EBA\u78BA\u8A8D\u304C\u5FC5\u8981\u3067\u3059\u3002",
  login_method: "\u30ED\u30B0\u30A4\u30F3\u65B9\u6CD5",
  ip: "IP",
  country: "\u56FD\u30FB\u5730\u57DF",
  city: "\u3068\u3057",
  sign_in_request: "\u30ED\u30B0\u30A4\u30F3\u8981\u6C42",
  requesting_sign_4361: "\u3053\u306E\u30A2\u30D7\u30EA\u30B1\u30FC\u30B7\u30E7\u30F3\u306F\u30ED\u30B0\u30A4\u30F3\u3092\u8981\u6C42\u3057\u307E\u3059",
  sign_in: "\u30ED\u30B0\u30A4\u30F3",
  copied_to: "\u30AF\u30EA\u30C3\u30D7\u30DC\u30FC\u30C9\u306B\u30B3\u30D4\u30FC",
  click_to_link_with_passkeys: "PassKeys\u3092\u4F7F\u7528\u3057\u305F\u30ED\u30B0\u30A4\u30F3",
  opening_passkeys: "\u30D0\u30A4\u30F3\u30C9PassKeys"
};
var account = {
  account_and_security: "\u30A2\u30AB\u30A6\u30F3\u30C8\u3068\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3",
  account: "\u30A2\u30AB\u30A6\u30F3\u30C8",
  verification_security: "\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u691C\u8A3C",
  verification_security_tip: "\u6B21\u306E\u65B9\u6CD5\u3092\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u691C\u8A3C\u306B\u4F7F\u7528\u3057\u307E\u3059\u3002",
  email: "Email",
  mobile: "\u643A\u5E2F\u756A\u53F7",
  payment_security: "\u652F\u6255\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3",
  payment_password: "\u652F\u6255\u3044\u30D1\u30B9\u30EF\u30FC\u30C9",
  set: "\u8A2D\u5B9A",
  pay_face_id: "FaceID\u3092\u4F7F\u7528\u3057\u305F\u652F\u6255\u3044",
  change_payment_password: "\u652F\u6255\u3044\u30D1\u30B9\u30EF\u30FC\u30C9\u306E\u5909\u66F4",
  forgot_payment_password: "\u30D1\u30B9\u30EF\u30FC\u30C9\u306E\u652F\u6255\u3044\u5FD8\u308C",
  set_your_email: "\u30E1\u30FC\u30EB\u30DC\u30C3\u30AF\u30B9\u306E\u8A2D\u5B9A",
  set_your_mobile: "\u643A\u5E2F\u96FB\u8A71\u756A\u53F7\u306E\u8A2D\u5B9A",
  set_your_email_tip: "\u3053\u306E\u30E1\u30FC\u30EB\u30DC\u30C3\u30AF\u30B9\u3092\u4F7F\u7528\u3057\u3066\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u691C\u8A3C\u3092\u884C\u3046\u3053\u3068\u304C\u3067\u304D\u307E\u3059\u3002",
  set_your_mobile_tip: "\u3053\u306E\u643A\u5E2F\u96FB\u8A71\u756A\u53F7\u3092\u4F7F\u7528\u3057\u3066\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u691C\u8A3C\u3092\u884C\u3046\u3053\u3068\u304C\u3067\u304D\u307E\u3059\u3002",
  input_vaild_email: "\u30E1\u30FC\u30EB\u30A2\u30C9\u30EC\u30B9\u3092\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044",
  input_vaild_mobile: "\u643A\u5E2F\u756A\u53F7\u3092\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044",
  switch_text_mobile: "\u643A\u5E2F\u96FB\u8A71\u756A\u53F7\u306E\u8A2D\u5B9A",
  switch_text_email: "\u30E1\u30FC\u30EB\u30DC\u30C3\u30AF\u30B9\u306E\u8A2D\u5B9A",
  place_email_address: "\u30E1\u30FC\u30EB\u30A2\u30C9\u30EC\u30B9",
  enter_code: "\u8A8D\u8A3C\u30B3\u30FC\u30C9\u306E\u5165\u529B",
  set_payment_password: "\u652F\u6255\u30D1\u30B9\u30EF\u30FC\u30C9\u306E\u8A2D\u5B9A",
  set_payment_password_tip: "\u652F\u6255\u3044\u8A8D\u8A3C\u7528\u306E\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u8A2D\u5B9A\u3057\u307E\u3059\u3002",
  re_enter_confirm: "\u78BA\u8A8D\u7528\u306B\u518D\u5165\u529B",
  done: "\u5B8C\u4E86",
  password_do_not_match: "2\u56DE\u5165\u529B\u3055\u308C\u305F\u30D1\u30B9\u30EF\u30FC\u30C9\u304C\u4E00\u81F4\u3057\u307E\u305B\u3093",
  payment_checking: "\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u30C1\u30A7\u30C3\u30AF\u3057\u3066\u3044\u307E\u3059",
  incorrect_tip1: "\u652F\u6255\u3044\u30D1\u30B9\u30EF\u30FC\u30C9\u304C\u6B63\u3057\u304F\u3042\u308A\u307E\u305B\u3093\u3002\u518D\u8A66\u884C\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  forgot_password: "\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u5FD8\u308C\u308B",
  retry: "\u518D\u8A66\u884C",
  change_enter_payment: "\u652F\u6255\u30D1\u30B9\u30EF\u30FC\u30C9\u306E\u5165\u529B",
  incorrect_payment_password: "\u652F\u6255\u3044\u30D1\u30B9\u30EF\u30FC\u30C9\u304C\u6B63\u3057\u304F\u3042\u308A\u307E\u305B\u3093",
  incorrect_tip2: "\u3042\u306A\u305F\u306F\u9593\u9055\u3063\u305F\u30D1\u30B9\u30EF\u30FC\u30C9\u309210\u56DE\u5165\u529B\u3057\u307E\u3057\u305F\u300224\u6642\u9593\u5F85\u3063\u3066\u304B\u3089\u518D\u8A66\u884C\u3059\u308B\u304B\u3001\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u5FD8\u308C\u3066\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u30EA\u30BB\u30C3\u30C8\u3059\u308B\u3092\u30AF\u30EA\u30C3\u30AF\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  modal_select_verification: "\u691C\u8A3C\u65B9\u6CD5\u306E\u9078\u629E",
  waring_tip1: "\u4ECA\u3059\u3050\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u30EC\u30D9\u30EB\u3092\u5411\u4E0A\uFF01\u6697\u53F7\u5316\u3055\u308C\u305F\u8CC7\u7523\u3092\u3088\u308A\u4FDD\u8B77\u3059\u308B\u305F\u3081\u306B\u652F\u6255\u3044\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u8A2D\u5B9A\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  account_modal1_title: "\u3042\u306A\u305F\u306F\u307E\u3060\u652F\u6255\u3044\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u8A2D\u5B9A\u3057\u3066\u3044\u307E\u305B\u3093",
  account_modal1_tip: "\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u8A2D\u5B9A\u3059\u308B\u3053\u3068\u3067\u3001\u8CA1\u5E03\u306E\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u30EC\u30D9\u30EB\u3092\u52B9\u679C\u7684\u306B\u9AD8\u3081\u308B\u3053\u3068\u304C\u3067\u304D\u307E\u3059\u3002",
  account_modal1_btn1: "\u73FE\u5728\u8A2D\u5B9A",
  account_modal1_btn2: "\u8A2D\u5B9A\u3092\u4FDD\u7559\u3057\u3066\u7F72\u540D\u3092\u7D9A\u884C",
  unlock_face_id_tip: "Face ID\u3092\u30ED\u30C3\u30AF\u89E3\u9664\u3059\u308B\u305F\u3081\u306B\u652F\u6255\u3044\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u8A2D\u5B9A\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  open_face_id_tip: "FaceID\u3092\u30AA\u30F3\u306B\u3057\u3066\u3001\u3088\u308A\u826F\u3044\u652F\u6255\u3044\u4F53\u9A13\u3092\u697D\u3057\u3080\u3053\u3068\u304C\u3067\u304D\u307E\u3059\u3002",
  open_face_id: "Face ID\u3092\u958B\u304F",
  later: "\u8A2D\u5B9A\u3092\u4FDD\u7559\u4E2D",
  payment_password_hint: "\u9001\u91D1\u3084\u7F72\u540D\u306E\u78BA\u8A8D\u306B\u4F7F\u7528\u3055\u308C\u307E\u3059\u3002",
  learn_more: "\u8A73\u7D30",
  learn_more_period: "\u8A73\u7D30",
  master_password_tip: "\u8CC7\u7523\u306E\u5B89\u5168\u3092\u78BA\u4FDD\u3059\u308B\u305F\u3081\u306B\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u8A2D\u5B9A\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  master_password: "\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9",
  new_master_password: "\u65B0\u3057\u3044\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9",
  change_master_password: "\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u306E\u5909\u66F4",
  input_master_password: "\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044",
  current_master_password: "\u73FE\u5728\u306E\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9",
  password_error: "\u30D1\u30B9\u30EF\u30FC\u30C9\u30A8\u30E9\u30FC",
  about_master_password: "\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u306B\u3064\u3044\u3066",
  about_master_password_t1: "\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u3068\u306F\u4F55\u3067\u3059\u304B\uFF1F",
  about_master_password_p1: "\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u306F\u3001\u30ED\u30FC\u30AB\u30EB\u306E\u6697\u53F7\u5316\u30D5\u30E9\u30B0\u30E1\u30F3\u30C8\u3092\u6697\u53F7\u5316\u3059\u308B\u305F\u3081\u306B\u4F7F\u7528\u3055\u308C\u307E\u3059\u3002\u3053\u308C\u306F\u30A2\u30AB\u30A6\u30F3\u30C8\u3067\u6700\u3082\u91CD\u8981\u306A\u30D1\u30B9\u30EF\u30FC\u30C9\u3067\u3059\u3002Particle \u306F\u3053\u306E\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u4FDD\u5B58\u3057\u306A\u3044\u305F\u3081\u3001\u7D1B\u5931\u3057\u305F\u5834\u5408\u306F\u5FA9\u5143\u3067\u304D\u307E\u305B\u3093\u3002\u3053\u306E\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u9069\u5207\u306B\u4FDD\u7BA1\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  about_master_password_t2: "\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u306F\u3069\u306E\u3088\u3046\u306B\u4FDD\u8B77\u3057\u307E\u3059\u304B\uFF1F",
  about_master_password_p2: "\u65B0\u3057\u3044\u30C7\u30D0\u30A4\u30B9\u306B\u30ED\u30B0\u30A4\u30F3\u3059\u308B\u3068\u304D\u306B\u3001\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u5165\u529B\u3057\u3066\u6697\u53F7\u5316\u30D5\u30E9\u30B0\u30E1\u30F3\u30C8\u3092\u30A2\u30AF\u30C6\u30A3\u30D6\u306B\u3059\u308B\u5FC5\u8981\u304C\u3042\u308A\u307E\u3059\u3002\u305D\u3046\u3067\u306A\u3044\u5834\u5408\u3001\u30A6\u30A9\u30EC\u30C3\u30C8\u306B\u30A2\u30AF\u30BB\u30B9\u3067\u304D\u307E\u305B\u3093\u3002",
  about_master_password_t3: "Particle \u306E MPC-TSS \u30A6\u30A9\u30EC\u30C3\u30C8\u306B\u3064\u3044\u3066",
  about_master_password_p3: "Particle \u306F\u3001MPC\uFF08\u30DE\u30EB\u30C1\u30D1\u30FC\u30C6\u30A3\u8A08\u7B97\uFF09\u306B\u57FA\u3065\u304F TSS\uFF08\u95BE\u5024\u7F72\u540D\u30B9\u30AD\u30FC\u30E0\uFF09\u6697\u53F7\u30A6\u30A9\u30EC\u30C3\u30C8\u3092\u63D0\u4F9B\u3057\u3066\u3044\u307E\u3059\u3002\u3053\u308C\u306F\u3001\u53D6\u5F15\u3092\u958B\u59CB\u3059\u308B\u3068\u304D\u306B\u3001\u30C7\u30D0\u30A4\u30B9\u306E\u30ED\u30FC\u30AB\u30EB\u6697\u53F7\u5316\u30D5\u30E9\u30B0\u30E1\u30F3\u30C8\u3068 Particle \u306E\u6697\u53F7\u5316\u30D5\u30E9\u30B0\u30E1\u30F3\u30C8\u304C\u540C\u6642\u306B\u7F72\u540D\u3057\u3001\u305D\u308C\u304C\u30D6\u30ED\u30C3\u30AF\u30C1\u30A7\u30FC\u30F3\u30CD\u30C3\u30C8\u30EF\u30FC\u30AF\u306B\u9001\u4FE1\u3055\u308C\u308B\u3053\u3068\u3092\u610F\u5473\u3057\u307E\u3059\u3002\u3053\u308C\u306B\u3088\u308A\u3001\u30A6\u30A9\u30EC\u30C3\u30C8\u306E\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u304C\u5F37\u5316\u3055\u308C\u3001Shamir \u79D8\u5BC6\u5171\u6709\u30B9\u30AD\u30FC\u30E0\u306A\u3069\u306E\u4ED6\u306E\u65B9\u6CD5\u306B\u6BD4\u3079\u3066\u30C8\u30E9\u30F3\u30B6\u30AF\u30B7\u30E7\u30F3\u304C\u3088\u308A\u9AD8\u901F\u306B\u306A\u308A\u307E\u3059\u3002\u3055\u3089\u306B\u3001\u7F72\u540D\u30D7\u30ED\u30BB\u30B9\u306B\u95A2\u9023\u3059\u308B\u624B\u6570\u6599\u3082\u524A\u6E1B\u3055\u308C\u307E\u3059\u3002",
  set_master_password: "\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u306E\u8A2D\u5B9A",
  set_new_master_password: "\u65B0\u3057\u3044\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u306E\u8A2D\u5B9A",
  set_master_password_tip: "Particle\u306F\u3001\u8CC7\u7523\u306E\u5B89\u5168\u3092\u78BA\u4FDD\u3059\u308B\u305F\u3081\u306B\u3001\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u4F7F\u7528\u3057\u3066\u30C7\u30D0\u30A4\u30B9\u4E0A\u306E\u30C7\u30FC\u30BF\u3092\u6697\u53F7\u5316\u3057\u307E\u3059\u3002",
  set_master_password_now: "\u4ECA\u3059\u3050\u8A2D\u5B9A",
  restore_wallet: "\u8CA1\u5E03\u3092\u5143\u306B\u623B\u3059",
  mpc_tss_intro: "Particle\u30A6\u30A9\u30EC\u30C3\u30C8\u306F\u3001\u6697\u53F7\u5316\u3055\u308C\u305F\u8CC7\u7523\u3092\u4FDD\u8B77\u3059\u308B\u305F\u3081\u306BMPC-TSS\u3092\u4F7F\u7528\u3057\u307E\u3059\u3002",
  input_decrypt_hint: "\u6697\u53F7\u5316\u30B9\u30E9\u30A4\u30B9\u304C\u30C0\u30A6\u30F3\u30ED\u30FC\u30C9\u3055\u308C\u307E\u3057\u305F\u3002\u5FA9\u53F7\u3059\u308B\u305F\u3081\u306B\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  setting_successed: "\u8A2D\u5B9A\u306B\u6210\u529F\u3057\u307E\u3057\u305F",
  master_password_successed: "\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u306E\u8A2D\u5B9A\u306B\u6210\u529F\u3057\u307E\u3057\u305F\u3002\u30B9\u30E9\u30A4\u30B9\u306F\u6697\u53F7\u5316\u3055\u308C\u3066\u3044\u307E\u3059\u3002",
  master_password_store_hint: "Particle\u306F\u3053\u306E\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u4FDD\u5B58\u3057\u307E\u305B\u3093\u3002\u5FD8\u308C\u306A\u3044\u3088\u3046\u306B\u3057\u3066\u304F\u3060\u3055\u3044\u3002\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u7D1B\u5931\u3059\u308B\u3068\u3001\u8CC7\u91D1\u306B\u30A2\u30AF\u30BB\u30B9\u3067\u304D\u306A\u304F\u306A\u308A\u307E\u3059\u3002",
  password_digits_hint: "\u30D1\u30B9\u30EF\u30FC\u30C9\u306F6\uFF5E20\u6587\u5B57\u3067\u3042\u308B\u5FC5\u8981\u304C\u3042\u308A\u307E\u3059\u3002",
  confirm_master_password: "\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u306E\u78BA\u8A8D",
  confirm_new_master_password: "\u65B0\u3057\u3044\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u306E\u78BA\u8A8D",
  master_password_not_match: "\u30D1\u30B9\u30EF\u30FC\u30C9\u304C\u4E00\u81F4\u3057\u307E\u305B\u3093",
  master_password_use1: "Particle\u306F\u3001\u8CC7\u7523\u306E\u5B89\u5168\u3092\u78BA\u4FDD\u3059\u308B\u305F\u3081\u306B\u3001\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u4F7F\u7528\u3057\u3066\u30C7\u30D0\u30A4\u30B9\u4E0A\u306E\u30C7\u30FC\u30BF\u3092\u6697\u53F7\u5316\u3057\u307E\u3059\u3002",
  master_password_note: "\u6CE8\uFF1A\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u5909\u66F4\u3057\u305F\u5834\u5408\u3001\u4ED6\u306E\u30C7\u30D0\u30A4\u30B9\u3067\u3082\u30A6\u30A9\u30EC\u30C3\u30C8\u3092\u4F7F\u7528\u3059\u308B\u305F\u3081\u306B\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u306E\u518D\u78BA\u8A8D\u304C\u5FC5\u8981\u306B\u306A\u308A\u307E\u3059\u3002",
  master_password_set_failed: "\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u8A2D\u5B9A\u3067\u304D\u307E\u305B\u3093\u3002\u30CD\u30C3\u30C8\u30EF\u30FC\u30AF\u3092\u78BA\u8A8D\u3057\u3066\u518D\u8A66\u884C\u3057\u3066\u304F\u3060\u3055\u3044\u3002",
  master_password_loading_tip1: "\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u8A2D\u5B9A\u3057\u3066\u3044\u307E\u3059...",
  master_password_loading_tip2: "\u30ED\u30FC\u30AB\u30EB\u30C7\u30FC\u30BF\u3092\u6697\u53F7\u5316\u3057\u3066\u3044\u307E\u3059...",
  master_password_loading_tip3: "\u30DA\u30FC\u30B8\u3092\u9589\u3058\u306A\u3044\u3067\u304F\u3060\u3055\u3044...",
  more_login_account: "\u30EA\u30F3\u30AF",
  login_account: "\u30ED\u30B0\u30A4\u30F3\u65B9\u6CD5",
  login_account_hint: "\u30ED\u30B0\u30A4\u30F3\u7528\u306B\u4ED6\u306E\u30A2\u30AB\u30A6\u30F3\u30C8\u3092\u30D0\u30A4\u30F3\u30C9\u3059\u308B\u3053\u3068\u306F\u3067\u304D\u307E\u3059\u304C\u3001\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u30A2\u30AB\u30A6\u30F3\u30C8\u3068\u540C\u3058\u306B\u3059\u308B\u3053\u3068\u306F\u3067\u304D\u307E\u305B\u3093\u3002",
  not_linked: "\u30D0\u30A4\u30F3\u30C9\u3055\u308C\u3066\u3044\u307E\u305B\u3093",
  unlink_hint: "\u3053\u306E\u30A2\u30AB\u30A6\u30F3\u30C8\u3092\u4F7F\u7528\u3057\u3066\u306B\u30ED\u30B0\u30A4\u30F3\u3067\u304D\u307E\u3059",
  unlink: "\u30D0\u30A4\u30F3\u30C9\u89E3\u9664",
  link: "\u30D0\u30A4\u30F3\u30C9",
  close_payment_password: "\u652F\u6255\u3044\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u9589\u3058\u308B",
  close_success: "\u6B63\u5E38\u306B\u7D42\u4E86\u3057\u307E\u3057\u305F",
  security: "\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3",
  security_account_bind_hint: "\u30A2\u30AB\u30A6\u30F3\u30C8\u306E\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u8A8D\u8A3C\u3092\u8A2D\u5B9A\u3057\u305F\u5F8C\u3001\u3053\u306E\u6A5F\u80FD\u3092\u4F7F\u7528\u3067\u304D\u307E\u3059\u3002",
  unset_the_master_password: "\u7D42\u4E86\u3057\u3066\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u8A2D\u5B9A\u3057\u307E\u3059\u304B\uFF1F\u30DE\u30B9\u30BF\u30FC\u30D1\u30B9\u30EF\u30FC\u30C9\u304C\u8A2D\u5B9A\u3055\u308C\u3066\u3044\u306A\u3044\u3068\u30ED\u30B0\u30A4\u30F3\u306B\u5931\u6557\u3057\u307E\u3059\u3002",
  exit: "\u7D42\u4E86",
  create_passkeys: "PassKeys\u306E\u4F5C\u6210",
  create_at: "\u4F5C\u6210\uFF03\u30B5\u30AF\u30BB\u30A4\uFF03",
  username: "\u30E6\u30FC\u30B6\u30FC\u540D"
};
var risk = {
  transaction_risk: "\u53D6\u5F15\u30EA\u30B9\u30AF",
  malicious_address_detection: "\u30A2\u30C9\u30EC\u30B9\u306E\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u30C1\u30A7\u30C3\u30AF",
  nft_security_detection: "NFT\u306E\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u30C1\u30A7\u30C3\u30AF",
  token_security_detection: "\u30C8\u30FC\u30AF\u30F3\u306E\u30BB\u30AD\u30E5\u30EA\u30C6\u30A3\u30C1\u30A7\u30C3\u30AF",
  is_not_open_source: "\u3053\u306E\u5951\u7D04\u306F\u30AA\u30FC\u30D7\u30F3\u30BD\u30FC\u30B9\u3067\u306F\u3042\u308A\u307E\u305B\u3093",
  is_fake_token: "\u3053\u306E\u30C8\u30FC\u30AF\u30F3\u306F\u507D\u9020\u3055\u308C\u305F\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  nft_not_verified: "\u3053\u306ENFT\u5951\u7D04\u306F\u691C\u8A3C\u3055\u308C\u3066\u3044\u307E\u305B\u3093",
  nft_not_open_source: "\u3053\u306ENFT\u5951\u7D04\u306F\u30AA\u30FC\u30D7\u30F3\u30BD\u30FC\u30B9\u3067\u306F\u3042\u308A\u307E\u305B\u3093",
  nft_verified: "\u3053\u306ENFT\u306F\u4E3B\u8981\u306ANFT\u30D7\u30E9\u30C3\u30C8\u30D5\u30A9\u30FC\u30E0\u3067\u8A8D\u8A3C\u3055\u308C\u3066\u3044\u307E\u305B\u3093",
  same_nfts: "\u3053\u306ENFT\u306F\u91CD\u8907\u3057\u3066\u3044\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  nft_open_source: "\u3053\u306ENFT\u306F\u30AA\u30FC\u30D7\u30F3\u30BD\u30FC\u30B9\u3067\u306F\u3042\u308A\u307E\u305B\u3093",
  nft_proxy: "\u3053\u306ENFT\u306B\u306F\u5951\u7D04\u30D7\u30ED\u30AD\u30B7\u304C\u3042\u308A\u307E\u305B\u3093",
  privileged_burn: "\u3053\u306ENFT\u306F\u5951\u7D04\u6240\u6709\u8005\u306B\u3088\u3063\u3066\u7834\u68C4\u3055\u308C\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  transfer_without_approval: "\u3053\u306ENFT\u306F\u5951\u7D04\u6240\u6709\u8005\u306B\u3088\u3063\u3066\u4ED6\u306E\u30A2\u30C9\u30EC\u30B9\u306B\u79FB\u52D5\u3067\u304D\u307E\u3059",
  privileged_minting: "\u3053\u306ENFT\u5951\u7D04\u306B\u306F\u5897\u767A\u6A5F\u80FD\u304C\u5B58\u5728\u3059\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  self_destruct: "\u3053\u306ENFT\u5951\u7D04\u306B\u306F\u81EA\u5DF1\u7834\u58CA\u6A5F\u80FD\u304C\u5B58\u5728\u3057\u307E\u3059",
  restricted_approval: "\u4E00\u822C\u30E6\u30FC\u30B6\u30FC\u306F\u3053\u306ENFT\u3092\u6B63\u5E38\u306B\u53D6\u5F15\u3067\u304D\u306A\u3044\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  oversupply_minting: "\u3053\u306ENFT\u5951\u7D04\u306B\u306F\u5897\u767A\u6A5F\u80FD\u304C\u5B58\u5728\u3059\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  is_open_source: "\u3053\u306E\u30C8\u30FC\u30AF\u30F3\u5951\u7D04\u306F\u30AA\u30FC\u30D7\u30F3\u30BD\u30FC\u30B9\u3067\u306F\u3042\u308A\u307E\u305B\u3093",
  is_proxy: "\u3053\u306E\u30C8\u30FC\u30AF\u30F3\u5951\u7D04\u306B\u306F\u30D7\u30ED\u30AD\u30B7\u30E1\u30AB\u30CB\u30BA\u30E0\u304C\u542B\u307E\u308C\u3066\u3044\u307E\u305B\u3093",
  is_mintable: "\u3053\u306E\u30C8\u30FC\u30AF\u30F3\u5951\u7D04\u306B\u306F\u5897\u767A\u6A5F\u80FD\u304C\u5B58\u5728\u3059\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  can_take_back_ownership: "\u3053\u306E\u30C8\u30FC\u30AF\u30F3\u5951\u7D04\u306B\u306F\u6240\u6709\u6A29\u306E\u53D6\u308A\u623B\u3057\u30ED\u30B8\u30C3\u30AF\u304C\u5B58\u5728\u3057\u307E\u3059",
  owner_change_balance: "\u3053\u306E\u30C8\u30FC\u30AF\u30F3\u306F\u5951\u7D04\u6240\u6709\u8005\u306B\u3088\u3063\u3066\u30C8\u30FC\u30AF\u30F3\u306E\u6B8B\u9AD8\u304C\u5909\u66F4\u3055\u308C\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  hidden_owner: "\u3053\u306E\u30C8\u30FC\u30AF\u30F3\u306B\u306F\u96A0\u3055\u308C\u305F\u30AA\u30FC\u30CA\u30FC\u30A2\u30C9\u30EC\u30B9\u304C\u5B58\u5728\u3057\u307E\u3059",
  external_call: "\u3053\u306E\u5951\u7D04\u306E\u5B9F\u884C\u6642\u306B\u5916\u90E8\u5951\u7D04\u304C\u547C\u3073\u51FA\u3055\u308C\u307E\u3059",
  selfdestruct: "\u30C8\u30FC\u30AF\u30F3\u5951\u7D04\u306B\u306F\u81EA\u5DF1\u7834\u58CA\u6A5F\u80FD\u304C\u5B58\u5728\u3057\u307E\u3059",
  buy_tax: "\u7A0E\u7387\u304C10\uFF05\u3092\u8D85\u3048\u308B\u3068\u9AD8\u3044\u3068\u3055\u308C\u300150\uFF05\u3092\u8D85\u3048\u308B\u3068\u53D6\u5F15\u304C\u3067\u304D\u306A\u304F\u306A\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  sell_tax: "\u7A0E\u7387\u304C10\uFF05\u3092\u8D85\u3048\u308B\u3068\u9AD8\u3044\u3068\u3055\u308C\u300150\uFF05\u3092\u8D85\u3048\u308B\u3068\u53D6\u5F15\u304C\u3067\u304D\u306A\u304F\u306A\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  is_honeypot: "\u3053\u306E\u30C8\u30FC\u30AF\u30F3\u306F\u8C94\u8C85\u30C8\u30FC\u30AF\u30F3\u307E\u305F\u306F\u60AA\u610F\u306E\u3042\u308B\u30B3\u30FC\u30C9\u3092\u542B\u3093\u3067\u3044\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u3001\u30E6\u30FC\u30B6\u30FC\u304C\u8CFC\u5165\u3057\u305F\u5F8C\u3001\u58F2\u5374\u3067\u304D\u306A\u3044\u304B\u91CD\u5927\u306A\u640D\u5931\u3092\u88AB\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  transfer_pausable: "\u5951\u7D04\u306B\u306F\u53D6\u5F15\u3092\u4E00\u6642\u505C\u6B62\u3059\u308B\u30ED\u30B8\u30C3\u30AF\u304C\u5B58\u5728\u3057\u3001\u30C8\u30FC\u30AF\u30F3\u306E\u58F2\u8CB7\u53D6\u5F15\u304C\u5B8C\u5168\u306B\u4E00\u6642\u505C\u6B62\u3055\u308C\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  trading_cooldown: "\u3053\u306E\u30C8\u30FC\u30AF\u30F3\u5951\u7D04\u306B\u306F\u53D6\u5F15\u306E\u51B7\u5374\u6A5F\u80FD\u304C\u5B58\u5728\u3057\u307E\u3059\u3002\u8CFC\u5165\u5F8C\u306E\u4E00\u5B9A\u6642\u9593\u307E\u305F\u306F\u30D6\u30ED\u30C3\u30AF\u5185\u3067\u306F\u30C8\u30FC\u30AF\u30F3\u3092\u58F2\u5374\u3067\u304D\u307E\u305B\u3093",
  is_anti_whale: "\u3053\u306E\u30C8\u30FC\u30AF\u30F3\u306E\u58F2\u5374\u6570\u91CF\u306B\u5236\u9650\u304C\u3042\u308A\u307E\u3059",
  anti_whale_modifiable: "\u53D6\u5F15\u306E\u6700\u5927\u91D1\u984D\u307E\u305F\u306F\u6700\u5927\u4FDD\u6709\u91CF\u304C\u5909\u66F4\u53EF\u80FD\u3067\u3042\u308A\u3001\u53D6\u5F15\u304C\u4E00\u6642\u505C\u6B62\u3059\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  cannot_buy: "\u3053\u306E\u30C8\u30FC\u30AF\u30F3\u306FdApp\u306E\u5831\u916C\u3068\u3057\u3066\u767A\u884C\u3055\u308C\u3066\u304A\u308A\u3001\u30E6\u30FC\u30B6\u30FC\u306F\u76F4\u63A5\u8CFC\u5165\u3059\u308B\u3053\u3068\u306F\u3067\u304D\u307E\u305B\u3093",
  cannot_sell_all: "\u3053\u306E\u30C8\u30FC\u30AF\u30F3\u306F\u4FDD\u6709\u3057\u3066\u3044\u308B\u30C8\u30FC\u30AF\u30F3\u3092\u5B8C\u5168\u306B\u58F2\u5374\u3059\u308B\u3053\u3068\u306F\u3067\u304D\u307E\u305B\u3093",
  slippage_modifiable: "\u3053\u306E\u30C8\u30FC\u30AF\u30F3\u306E\u30D7\u30ED\u30B8\u30A7\u30AF\u30C8\u306F\u53D6\u5F15\u7A0E\u3092\u5909\u66F4\u3059\u308B\u6A29\u9650\u3092\u4FDD\u6301\u3057\u3066\u3044\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  is_blacklisted: "\u5951\u7D04\u306B\u306F\u30D6\u30E9\u30C3\u30AF\u30EA\u30B9\u30C8\u30E1\u30AB\u30CB\u30BA\u30E0\u304C\u542B\u307E\u308C\u3066\u304A\u308A\u3001\u4E00\u90E8\u306E\u30E6\u30FC\u30B6\u30FC\u304C\u53D6\u5F15\u5236\u9650\u3092\u53D7\u3051\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  is_whitelisted: "\u5951\u7D04\u306B\u306F\u30DB\u30EF\u30A4\u30C8\u30EA\u30B9\u30C8\u30E1\u30AB\u30CB\u30BA\u30E0\u304C\u542B\u307E\u308C\u3066\u304A\u308A\u3001\u4E00\u90E8\u306E\u30E6\u30FC\u30B6\u30FC\u304C\u53D6\u5F15\u5236\u9650\u3092\u53D7\u3051\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  personal_slippage_modifiable: "\u6307\u5B9A\u3057\u305F\u30A2\u30C9\u30EC\u30B9\u3054\u3068\u306B\u53D6\u5F15\u7A0E\u3092\u5909\u66F4\u3067\u304D\u3001\u30AA\u30FC\u30CA\u30FC\u306F\u6307\u5B9A\u3057\u305F\u30A2\u30C9\u30EC\u30B9\u306B\u975E\u5E38\u306B\u9AD8\u3044\u7A0E\u7387\u3092\u8A2D\u5B9A\u3057\u3066\u53D6\u5F15\u3092\u59A8\u5BB3\u3059\u308B\u3053\u3068\u304C\u3067\u304D\u307E\u3059",
  is_true_token: "\u3053\u306E\u30C8\u30FC\u30AF\u30F3\u306F\u91CD\u540D\u30C8\u30FC\u30AF\u30F3\u304B\u3082\u3057\u308C\u307E\u305B\u3093",
  is_airdrop_scam: "\u3053\u308C\u306F\u7A7A\u632F\u308A\u8A50\u6B3A\u30C8\u30FC\u30AF\u30F3\u3067\u3042\u308A\u3001\u3053\u306E\u30C8\u30FC\u30AF\u30F3\u306E\u516C\u5F0F\u30A6\u30A7\u30D6\u30B5\u30A4\u30C8\u306B\u6A29\u9650\u3092\u4E0E\u3048\u308B\u3068\u8CC7\u7523\u304C\u640D\u5931\u3059\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  other_potential_risks: "\u305D\u306E\u4ED6\u306E\u6F5C\u5728\u7684\u306A\u30EA\u30B9\u30AF",
  phishing_activities: "\u3053\u306E\u30A2\u30C9\u30EC\u30B9\u306F\u30D5\u30A3\u30C3\u30B7\u30F3\u30B0\u6D3B\u52D5\u306B\u95A2\u4E0E\u3057\u3066\u3044\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  blackmail_activities: "\u3053\u306E\u30A2\u30C9\u30EC\u30B9\u306F\u8EAB\u4EE3\u91D1\u8981\u6C42\u6D3B\u52D5\u306B\u95A2\u4E0E\u3057\u3066\u3044\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  stealing_attack: "\u3053\u306E\u30A2\u30C9\u30EC\u30B9\u306F\u30B3\u30A4\u30F3\u306E\u76D7\u96E3\u653B\u6483\u306B\u95A2\u4E0E\u3057\u3066\u3044\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  honeypot_related_address: "\u3053\u306E\u30A2\u30C9\u30EC\u30B9\u306F\u8C94\u8C85\u306A\u3069\u306E\u9AD8\u30EA\u30B9\u30AF\u30C8\u30FC\u30AF\u30F3\u306B\u95A2\u4E0E\u3057\u3066\u3044\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  fake_kyc: "\u3053\u306E\u30A2\u30C9\u30EC\u30B9\u306F\u507D\u306EKYC\u306B\u95A2\u4E0E\u3057\u3066\u3044\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  malicious_mining_activities: "\u3053\u306E\u30A2\u30C9\u30EC\u30B9\u306F\u60AA\u610F\u306E\u3042\u308B\u30DE\u30A4\u30CB\u30F3\u30B0\u6D3B\u52D5\u306B\u95A2\u4E0E\u3057\u3066\u3044\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  darkweb_transactions: "\u3053\u306E\u30A2\u30C9\u30EC\u30B9\u306F\u30C0\u30FC\u30AF\u30A6\u30A7\u30D6\u306E\u53D6\u5F15\u306B\u95A2\u4E0E\u3057\u3066\u3044\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  cybercrime: "\u3053\u306E\u30A2\u30C9\u30EC\u30B9\u306F\u30B5\u30A4\u30D0\u30FC\u72AF\u7F6A\u306B\u95A2\u4E0E\u3057\u3066\u3044\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  money_laundering: "\u3053\u306E\u30A2\u30C9\u30EC\u30B9\u306F\u30DE\u30CD\u30FC\u30ED\u30F3\u30C0\u30EA\u30F3\u30B0\u6D3B\u52D5\u306B\u95A2\u4E0E\u3057\u3066\u3044\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  financial_crime: "\u3053\u306E\u30A2\u30C9\u30EC\u30B9\u306F\u91D1\u878D\u72AF\u7F6A\u306B\u95A2\u4E0E\u3057\u3066\u3044\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  mixer: "\u3053\u306E\u30A2\u30C9\u30EC\u30B9\u306F\u30DF\u30AD\u30B7\u30F3\u30B0\u30B5\u30FC\u30D3\u30B9\u306B\u95A2\u4E0E\u3057\u3066\u3044\u308B\u53EF\u80FD\u6027\u304C\u3042\u308A\u307E\u3059",
  sanctioned: "\u3053\u306E\u30A2\u30C9\u30EC\u30B9\u306F\u7B2C\u4E09\u8005\u306E\u5236\u88C1\u3092\u53D7\u3051\u305F\u30A2\u30C9\u30EC\u30B9\u3067\u3059",
  blacklist_doubt: "\u305D\u306E\u4ED6\u306E\u60AA\u610F\u306E\u3042\u308B\u884C\u52D5"
};
var jp_default = {
  common,
  login,
  sign,
  error,
  program,
  new: new2,
  account,
  risk
};
export {
  account,
  common,
  jp_default as default,
  error,
  login,
  new2 as new,
  program,
  risk,
  sign
};
